pysolr.py

```diff
@@ -820,7 +829,16 @@ def search(self, q, search_handler=None, **kwargs):
             # cover both cases: there is no response key or value is None
             (decoded.get("response", {}) or {}).get("numFound", 0),
         )
-        return self.results_cls(decoded)
+
+        if "cursorMark" in params and params["cursorMark"] != \
+                decoded.get('nextCursorMark', params["cursorMark"]):
+            def next_page_query():
+                nextParams = dict(params)
```

Comment:
This might be more clear about the intention:
```suggestion
                nextParams = params.copy()
```